common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleServiceMetrics.java

```diff
@@ -78,7 +82,17 @@ public static void collectMetric(
           new ShuffleServiceMetricsInfo(name + "_rate1", "1 minute rate of timer " + name),
           t.getOneMinuteRate())
         .addGauge(new ShuffleServiceMetricsInfo(name + "_rateMean", "Mean rate of timer " + name),
-          t.getMeanRate());
+          t.getMeanRate())
+        .addGauge(getMetricsInfoGenericValue(timingName, "max"), snapshot.getMax())
+        .addGauge(getMetricsInfoGenericValue(timingName, "min"), snapshot.getMin())
+        .addGauge(getMetricsInfoGenericValue(timingName, "mean"), snapshot.getMean())
+        .addGauge(getMetricsInfoGenericValue(timingName, "stdDev"), snapshot.getStdDev());
+      for (int percentile : new int[] { 1, 5, 25, 50, 75, 95, 99 }) {
```

Comment:
Is there a reason of this selection? In general, we use `Snapshot.get98thPercentile` and `Snapshot.get999thPercentile` additionally too.

Reply:
I added 98th and 99.9th percentiles. Thanks for the input!

One thing that may seem unusual is the low percentiles (1, 5, 25), which have found useful internally for a getting a better sense of the full range of timing information. Let me know if you think they are too non-standard to include here.